### PR TITLE
Test/repository implementation

### PR DESCRIPTION
**Pull request by saraashraf11**
# Pull Request Template

## Description

This pull request introduces a major enhancement to the unit testing suite for the repository layer. The primary purpose is to increase code coverage, validate the correctness of data handling logic, and ensure the stability of our repositories against future changes.

---

## Changes Made

- Added New Unit Tests: Introduced a complete test suite for GameRepositoryImpl to ensure its logic is fully covered and validated.
- Enhanced Test Coverage: Added numerous new test cases to existing test suites for the following repositories, significantly increasing test coverage and verifying more edge cases:
      AppPreferencesRepositoryImplTest
      AuthenticationRepositoryImplTest
      CountryRepositoryImplTest
      MovieRepositoryImplTest
      RecentSearchRepositoryImplTest
      TvShowRepositoryImplTest
      WatchHistoryRepositoryImplTest

---

## Checklist
Please ensure the following tasks are completed:

- [x] My code follows the code style of this project
- [x] Changes have been tested manually and verified.
- [x] PR includes at most one single feature.

### CODE REVIEW

**Review by mohamedd-hassan — DISMISSED** [review]

**Review by Ahmed-AlMancy — DISMISSED** [review]

File: repository/src/test/java/com/amsterdam/repository/repository/AuthenticationRepositoryImplTest.kt [hunk 1]
```diff
@@ -49,132 +39,92 @@ class AuthenticationRepositoryImplTest {
...
+    fun `loginWithPassword should throw exception on remote failure`() = runTest {
+        val expectedException = RuntimeException("Remote login failed")
+        coEvery {
+            authenticationRemoteDataSource.loginWithPassword(
+                any(),
+                any()
+            )
+        } throws expectedException
+
+        val thrownException = assertThrows<RuntimeException> {
```
Comment (nabilBouzineDev) [inline repository/src/test/java/com/amsterdam/repository/repository/AuthenticationRepositoryImplTest.kt]: There's No need to have this in variable with expectedException. We can make assertion directly like this.

assertThrows<RuntimeException> {  repository.loginWithPassword("user", "pass")  }

File: repository/src/test/java/com/amsterdam/repository/repository/AuthenticationRepositoryImplTest.kt [hunk 2]
```diff
@@ -49,132 +39,92 @@ class AuthenticationRepositoryImplTest {
...
+        val expectedException = RuntimeException("Failed to read session type")
+        coEvery { authenticationLocalDataSource.getSessionType() } throws expectedException
 
-            // When
-            val result = repository.getSessionType()
-
-            // Then
-            assertThat(result).isEqualTo(SessionType.GUEST)
-            coVerify(exactly = 1) { authenticationLocalDataSource.getSessionType() }
+        val thrownException = assertThrows<RuntimeException> {
```
Comment (nabilBouzineDev) [inline repository/src/test/java/com/amsterdam/repository/repository/AuthenticationRepositoryImplTest.kt]: same issue here

**Review by nabilBouzineDev — CHANGES_REQUESTED** [review]

File: repository/src/test/java/com/amsterdam/repository/repository/AuthenticationRepositoryImplTest.kt [hunk 3]
```diff
@@ -49,132 +39,92 @@ class AuthenticationRepositoryImplTest {
...
-                authenticationRemoteDataSource.loginWithPassword(
-                    testUsername,
-                    testPassword
-                )
-            }
-            verify(exactly = 0) { cryptoManager.encryptString(any()) }
-            coVerify(exactly = 0) { authenticationLocalDataSource.cacheSessionId(any()) }
-            coVerify(exactly = 0) { authenticationLocalDataSource.setSessionType(any()) }
-        }
+        assertThat(thrownException).isEqualTo(expectedException)
```
Comment (eslam-99) [inline repository/src/test/java/com/amsterdam/repository/repository/AuthenticationRepositoryImplTest.kt]: There should be 1 assertion per test, so you can split this test into 2 tests, 1 has the assertion and the other has the coVerifiy
This can be applied to all tests you modified

**Review by Ahmed-AlMancy — APPROVED** [review]

**Review by nabilBouzineDev — APPROVED** [review]

**Review by ahmedmahgoub0 — APPROVED** [review]
